### ISSUE TITLE
NamedFile partial ranges not working correctly with Chrome

### ISSUE THREAD

**Issue by cetra3** [actix/actix-web#238]
Appears to be introduced by https://github.com/actix/actix-web/pull/227

Running Version `0.6.8`, appears as though NamedFile's partial ranges is causing a number of errors when trying to serve static files.

I have a handler for defined like so for a SPA:

```rust
pub struct FrontHandler {
    //front end assets
    front: String,
}


impl<S: 'static> Handler<S> for FrontHandler {
    type Result = Result<fs::NamedFile>;

    fn handle(&mut self, req: HttpRequest<S>) -> Self::Result {
        let mut relative_path = req.uri().path().replacen("/", "", 1);

        if relative_path == "" || relative_path.ends_with("/") {
            relative_path.push_str("index.html");
        }

        let final_path = format!("{}{}", self.front, relative_path);
        let path = PathBuf::from(final_path);

        debug!("Path:{:?}, exists:{}", path, path.exists());

        Ok(fs::NamedFile::open(&path)?)
    }
}
```

Using this with `0.6.8` I'm seeing the following errors in the logs:

``` 
ERROR actix_web::pipeline           > Error occurred during request handling: Last payload item, but eof is not reached
ERROR actix_web::server::h1         > Unhandled error: Last payload item, but eof is not reached
```

And I am seeing these errors like this in Chrome `66.0.3359.181` (after caching the files the first time):

![screen shot 2018-05-23 at 3 13 40 pm](https://user-images.githubusercontent.com/6415435/40405476-eee24a32-5e9b-11e8-9956-4e115db55f7e.png)

What chrome appears to be doing is sending a 0 length range request:

```
Range: bytes=53395-53395
```

And Actix is responding with in turn with the following:

```
content-length: 1
content-range: bytes 53395-53395/290992
```

But I believe the content length should be 0 in this case

**Comment by cetra3** [comment]
Appears to be resolved `0.6.9`, maybe close this one off

**Comment by max-frai** [comment]
Yes, I had the same issue and this made me to disable compression for some static files, now its resolved in last commits.

**Comment by fafhrd91** [comment]
Fix is released with 0.6.9
